### PR TITLE
upgrade to d3-v4 plugin structure

### PR DESCRIPTION
**Pull request by sly7-7**
related #107

### CODE REVIEW

**Comment by robodude** [comment]
Maybe this could be published to npm separately?

**Comment by sly7-7** [comment]
@robodude I've just publish [d3-v4-cloud ](https://www.npmjs.com/package/d3-v4-cloud). You can have a try.

**Comment by robodude** [comment]
Gave it a try and it seems to work quite well so far, thanks! :)

File: index.js [hunk 1]
```diff
@@ -1,399 +1,4 @@
...
-
-function functor(d) {
-  return typeof d === "function" ? d : function() { return d; };
-}
-
-var spirals = {
-  archimedean: archimedeanSpiral,
-  rectangular: rectangularSpiral
-};
+export {default as cloud} from "./src/d3-cloud";
```
Comment (aknuds1) [inline index.js]: Also, is there any reason for the export now being named (`cloud`), as opposed to just being the default? It breaks examples/browserify.js as this expects the plugin to be exported as default.

**Comment by jasondavies** [comment]
Is this a purely cosmetic change to the code structure, or is there a functional reason too?

**Comment by aknuds1** [comment]
@jasondavies Good question, I was wondering about the reason for moving the code too.

**Comment by simonmcl** [comment]
I'm quite new to d3 and angular, but i'm having a lot of trouble trying to integrate this package into Angular 2 with d3v4. Code samples online just keep telling me that "layout" can't be found on element d3 for this code:

`d3.layout.cloud()`

I'm guessing that this module system has been changed between d3v3 and d3v4? I think it would be worth re-looking at this pull request and updating the DefinitelyTyped types entry `d3.cloud.layout`

@sly7-7 would you be able to create an `@types` (e.g. https://www.npmjs.com/package/@types/d3.cloud.layout ) for your module as Angular2 doesn't know how to process it?

**Comment by sly7-7** [comment]
Unfortunately I m sick for now, and can't do anything (actually I don't
even have a computer). I could eventually take a look at this in two weeks
(I hope so...)

Le 2 févr. 2017 10:42, "Simon McLoughlin" <notifications@github.com> a
écrit :

**Pull request closed without merging**